crates/rpc/rpc/src/eth/api/traits/pending_block.rs

```diff
@@ -0,0 +1,144 @@
+//! Loads a pending block from database. Helper trait for `eth_` block and transaction RPC methods.
+
+use std::time::{Duration, Instant};
+
+use futures::Future;
+use reth_evm::ConfigureEvm;
+use reth_primitives::{SealedBlockWithSenders, SealedHeader};
+use reth_provider::{
+    BlockReader, BlockReaderIdExt, ChainSpecProvider, EvmEnvProvider, StateProviderFactory,
+};
+use reth_transaction_pool::TransactionPool;
+use revm_primitives::{BlockEnv, CfgEnv, CfgEnvWithHandlerCfg, SpecId};
+use tokio::sync::Mutex;
+use tracing::debug;
+
+use crate::eth::{
+    api::{
+        pending_block::{PendingBlock, PendingBlockEnv, PendingBlockEnvOrigin},
+        SpawnBlocking,
+    },
+    error::{EthApiError, EthResult},
+};
+
+/// Loads a pending block from database.
+pub trait LoadPendingBlock: SpawnBlocking {
```

Comment:
hmm, this works for now 

This bakes in a few abstractions so that it has a default function. how would a different ethapi change this behaviour?

for now this would work, but I think it would be better if this can be implemented by the network

Reply:
a different network eth api doesn't have to change this behaviour, but it's necessary to be in a trait so it can be called in another trait's default trait method impl

Reply:
gotcha, now I got it